src/handlers/private-notifcations/issue-comment-created.ts

```diff
@@ -0,0 +1,112 @@
+    }
+
+    const usernameToClaimUrls: Record<string, string> = parsePaymentComment(payload.comment.body);
+
+    let bot;
+    try {
+        bot = (await TelegramBotSingleton.initialize(context.env)).getBot();
+    } catch (er) {
+        logger.error(`Error getting bot instance`, { er });
+    }
+
+    if (!bot) {
+        throw new Error("Bot instance not found");
+    }
+
+    for (const [telegramId, user] of Object.entries(users)) {
+        if (!user.listeningTo?.length) {
+            continue;
+        }
+
+        if (user.listeningTo.length === 1) {
+            const trigger = user.listeningTo[0];
+            await handleCommentNotificationTrigger(trigger, user, usernameToClaimUrls, telegramId, bot, logger);
+        } else {
+            for (const trigger of user.listeningTo) {
+                await handleCommentNotificationTrigger(trigger, user, usernameToClaimUrls, telegramId, bot, logger);
+            }
+        }
+    }
+
+    return { status: 200, reason: "success" };
+}
+
+async function handleCommentNotificationTrigger(trigger: string, user: UserBank[0], usernameToClaimUrls: Record<string, string>, telegramId: string, bot: any, logger: any) {
+    try {
+        switch (trigger) {
+            case "payment":
+                for (const [username, claimUrl] of Object.entries(usernameToClaimUrls)) {
+                    if (user.githubUsername === username) {
+                        await handlePaymentNotification(username, claimUrl, telegramId, bot);
+                    }
+                }
+                break;
+            case "reminder":
+                break;
+            case "disqualification":
+                break;
+            default:
+                logger.error(`Unknown trigger key ${trigger}`);
+        }
+    } catch (er) {
+        logger.error(`Error sending message to ${telegramId}`, { er });
+    }
+}
+
+async function handlePaymentNotification(
+    username: string,
+    claimUrlBase64String: string,
+    telegramId: string,
+    bot: Bot
+) {
+    const message = `${username}, a task reward has been generated for you\\. You can claim it [here](https://pay\\.ubq\\.fi?claim=${claimUrlBase64String})`
+
+    let userPrivateChat;
+
+    try {
+        userPrivateChat = await bot?.api.getChat(telegramId)
+    } catch (er) {
+        logger.error(`Error getting chat for ${telegramId}`, { er })
+    }
+
+    if (!userPrivateChat) {
+        logger.error(`This user has not started a chat with the bot yet`, { telegramId })
+        return;
+    }
+
+    try {
+        await bot?.api.sendMessage(telegramId, message, { parse_mode: "MarkdownV2" })
+    } catch (er) {
+        logger.error(`Error sending message to ${telegramId}`, { er })
+    }
+}
+
+function parsePaymentComment(comment: string) {
+    const regex = /href="https:\/\/[^\/]+\/?\?claim=([A-Za-z0-9+/=]+)"[^>]*>\s*\[.*?\]\s*<\/a>\s*<\/h3>\s*<h6>\s*@([a-zA-Z0-9-_]+)\s*<\/h6>/g
```

Comment:
This seems brittle can't we read from database?

I see you later in the code that is executing that issue comment, created so perhaps this is appropriate

Reply:
> This seems brittle can't we read from database?

I thought permits were only added to the Supabase tables after it was claimed no? There has been some development in that area recently with rollups etc so I'm unsure to be honest mate.

It does seem quite brittle. I could just extract the base64 strings and then decode them and match them against our userbase (after I script a migration file to port the data out) of wallets instead of trying to capture the username here which is the part that makes it brittle imo.

Reply:
The permit storage thing has actually been broken for months as far as I see from the logs because of duplicate rows and columns, I can have a look and fix it. I remember that we had a script that would populate the permits table parsing all the old issues?

Reply:
https://github.com/ubiquity/.github/pull/112

This pull you mean? It's been awaiting review for some time. I can have another look at it but I felt at the time it was as good as I could get it given the variables with permits on issues.